### PR TITLE
Fix issue with custom option file uploading

### PR DESCRIPTION
**Pull request by nikolaevas**
### Description (*)
If `sys_temp_dir` and `upload_tmp_dir` directives in _php.ini_ are different then it causes exception when uploading a file to product custom options on the frontend:

> Path "%1" cannot be used with directory "%2"

### Manual testing scenarios (*)

1. Set different values for `sys_temp_dir` and `upload_tmp_dir` in _php.ini_.
2. In magento create simple, visible in catalog product.
3. Add new customizable option to that product with option type **File**.
4. On product page on frontend select file for custom option and press Add to cart button.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=21170) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @nikolaevas. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: lib/internal/Magento/Framework/Filesystem/DirectoryList.php [hunk 1]
```diff
@@ -97,6 +97,10 @@ public function __construct($root, array $config = [])
         $this->root = $this->normalizePath($root);
         $this->directories = static::getDefaultConfig();
         $this->directories[self::SYS_TMP] = [self::PATH => realpath(sys_get_temp_dir())];
```
Comment (VladimirZaets) [inline lib/internal/Magento/Framework/Filesystem/DirectoryList.php]: Hi @nikolaevas. Thank for collaboration. 
You can use ternary operator in this case.
Something like: 
```$this->directories[self::SYS_TMP] ? ini_get('upload_tmp_dir') ? [self::PATH => realpath($uploadTmpDir)] : [self::PATH => realpath(sys_get_temp_dir())]```

Comment (nikolaevas) [reply lib/internal/Magento/Framework/Filesystem/DirectoryList.php]: Hi @VladimirZaets. I have updated the code.

**Review by osrecio — APPROVED** [review]
Thanks @nikolaevas for your contribution.

I tested in my local enviroment and works like a charm.

**Comment by magento-engcom-team** [comment]
Hi @osrecio, thank you for the review.
[ENGCOM-4275](https://jira.corp.magento.com/browse/ENGCOM-4275) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@nikolaevas thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**